packages/input-text/src/TextInput.js

```diff
@@ -0,0 +1,165 @@
+import * as React from "react";
+import PropTypes from "prop-types";
+import Tooltip from "@crave/farmblocks-tooltip";
+import Image, { badgeSizes } from "@crave/farmblocks-image";
+
+import errorIconSrc from "./constants/errorIcon";
+import StyledLabel from "./styledComponents/TextInput";
+
+class TextInput extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      value: this.props.value,
+      invalid: this.props.invalid,
+      showTooltip: false
+    };
+
+    this.onChange = this.onChange.bind(this);
+    this.onInvalid = this.onInvalid.bind(this);
+    this.onMouseOver = this.onMouseOver.bind(this);
+    this.onMouseLeave = this.onMouseLeave.bind(this);
+  }
+
+  render() {
+    const {
+      label,
+      value,
+      onChange,
+      invalid,
+      validationErrors,
+      onInvalid,
+      tooltipText,
+      errorIconSrc,
+      onMouseOver,
+      onMouseLeave,
+      ...otherProps
+    } = this.props;
+    const containerProps = {
+      filled: this.state.value.length > 0,
+      invalid: this.state.invalid,
+      disabled: otherProps.disabled
+    };
+    const inputProps = {
+      value: this.state.value,
+      onChange: this.onChange,
+      onInvalid: this.onInvalid,
+      onMouseOver: this.onMouseOver,
+      onMouseLeave: this.onMouseLeave,
+      ...otherProps
+    };
+    return (
+      <StyledLabel {...containerProps}>
+        {this._renderInput(inputProps)}
+        {this._renderLabel(label)}
```

Comment:
Any special reason to render input before `label` after `input` and change it with `order` css?

Reply:
to save the use of the for property and generate unique ids with the label after the input we can switch to the focused color with css only

Reply:
didn't find `for` on label... But ok, thanks

Reply:
yes, I am not using `for` 

in my comment above, I've use "to save" meaning "to avoid"

Reply:
I have added a comment on the code explaining this choice, but if you think a different implementation is better let me know